eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -95,6 +115,102 @@ function GetPackageLookup($packageList) {
+    Add-Content -Path $readmePath -Value $mgmtTable -NoNewline
+  }
+}
+
+# Update the metadata table on attributes: author, ms.author, ms.service
+function update-metadata-table($absolutePath, $readmeName, $serviceName, $msService)
+{
+  $readmePath = Join-Path $absolutePath -ChildPath $readmeName
+  $readmeContent = Get-Content -Path $readmePath -Raw
+  $null = $readmeContent -match "---`n*(?<metadata>(.*`n)*)---`n*(?<content>(.*`n)*)"
+  $restContent = $Matches["content"]
+
+  $lang = $LanguageDisplayName
+  $metadataString = GenerateDocsMsMetadata -language $lang -serviceName $serviceName `
+    -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
+    -msService $msService
+  Set-Content -Path $readmePath -Value "$metadataString`n$restContent" -NoNewline
+}
+
+function generate-markdown-table($absolutePath, $readmeName, $packageInfo, $moniker) {
+  $content = "| Reference | Package | Source |`r`n|---|---|---|`r`n" 
+  # Here is the table, the versioned value will
+  foreach ($pkg in $packageInfo) {
+    if (!$pkg.VersionGA -and "latest" -eq $moniker) {
+      continue
+    }
+    if (!$pkg.VersionPreview -and "preview" -eq $moniker) {
+      continue
+    }
+    $repositoryLink = $RepositoryUri
+    $packageLevelReadme = &$GetPackageLevelReadmeFn -packageMetadata $pkg
+    $referenceLink = "[$($pkg.DisplayName)]($packageLevelReadme-readme.md)"
+    if (!(Test-Path (Join-Path $absolutePath -ChildPath "$packageLevelReadme-readme.md"))) {
+      $referenceLink = $pkg.DisplayName
+    }
+    $githubLink = $GithubUri
+    if ($pkg.PSObject.Members.Name -contains "FileMetadata") {
+      $githubLink = "$GithubUri/blob/main/$($pkg.FileMetadata.DirectoryPath)"
+    }
+    $line = "|$referenceLink|[$($pkg.Package)]($repositoryLink/$($pkg.Package))|[Github]($githubLink)|`r`n"
+    $content += $line
+  }
+  Set-Content -Path (Join-Path $absolutePath -ChildPath $readmeName) -Value $content -NoNewline
+}
+
+function generate-service-level-readme($readmeBaseName, $pathPrefix, $packageInfos, $serviceName) {
+  # Add ability to override
+  # Fetch the service readme name
+  $monikers = @("latest", "preview")
+  $msService = GetDocsMsService -packageInfo $packageInfos[0] -serviceName $serviceName
+  foreach($moniker in $monikers) {
+    $absolutePath = "$DocRepoLocation/$pathPrefix/$moniker/"
+    $serviceReadme = "$readmeBaseName.md"
+    $clientIndexReadme  = "$readmeBaseName-client-index.md"
+    $mgmtIndexReadme  = "$readmeBaseName-mgmt-index.md"
+    $clientPackageInfo = $servicePackages.Where({ 'client' -eq $_.Type -and 'true' -eq $_.New}) | Sort-Object -Property Package
+    if ($clientPackageInfo) {
+      generate-markdown-table -absolutePath "$absolutePath" -readmeName "$clientIndexReadme" -packageInfo $clientPackageInfo -moniker $moniker
+    }
+    $mgmtPackageInfo = $servicePackages.Where({ 'mgmt' -eq $_.Type -and 'true' -eq $_.New }) | Sort-Object -Property Package
```

Comment:
Do we really want to filter by `'true' -eq $_.New` here? This would leave off packages that are in the ToC but not marked as `New` in the metadata.

Reply:
The package we used in package.json filtered by `New` to `true`, and the final reference.yml takes the package.json info into consideration, so it is safe to filter New to true.

Also, we are flexible to have manually created readme. So if there is any edge case which we can not handle, we still have the ability to add manual readme to fix it.